packages/neon-core/src/tx/components/WitnessScope.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Used to constrain witness to specified scope
+ * The execution in vm of a script is kind of opaque, user could add scope on his signature to avoid abuse of signature
+ */
+export enum WitnessScope {
```

Comment:
Hi @snowypowers , I want to discuss the design for this class with you, If we come to a consensus, I will start to implement the unit test.

Reply:
We can also implement it as enum flags here.

Reply:
Please check the new design

Reply:
https://basarat.gitbooks.io/typescript/docs/enums.html

This has a nice example on enum flags and having static enum methods. We shouldnt need a full class to manage flags.

Reply:
Oh, didn't know that. Great.